The MySQL native test is failing with:

```
2020-08-27T09:22:59.8943464Z com.oracle.svm.core.util.UserError$UserException: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method com.mysql.cj.jdbc.AbandonedConnectionCleanupThread.uncheckedShutdown() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
2020-08-27T09:22:59.8944140Z To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
2020-08-27T09:22:59.8944304Z Detailed message:
2020-08-27T09:22:59.8944420Z Trace: 
2020-08-27T09:22:59.8944549Z 	at parsing io.quarkus.jdbc.mysql.runtime.MySQLRecorder$1.run(MySQLRecorder.java:21)
2020-08-27T09:22:59.8944698Z Call path from entry point to io.quarkus.jdbc.mysql.runtime.MySQLRecorder$1.run(): 
2020-08-27T09:22:59.8944850Z 	at io.quarkus.jdbc.mysql.runtime.MySQLRecorder$1.run(MySQLRecorder.java:21)
2020-08-27T09:22:59.8944995Z 	at sun.nio.ch.EPollPort$EventHandlerTask.run(EPollPort.java:300)
2020-08-27T09:22:59.8945128Z 	at java.lang.Thread.run(Thread.java:834)
2020-08-27T09:22:59.8945240Z 	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
2020-08-27T09:22:59.8945560Z 	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
2020-08-27T09:22:59.8945741Z 	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
2020-08-27T09:22:59.8945853Z 
2020-08-27T09:22:59.8945983Z 	at com.oracle.svm.core.util.UserError.abort(UserError.java:79)
2020-08-27T09:22:59.8946130Z 	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:217)
2020-08-27T09:22:59.8946265Z 	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:753)
2020-08-27T09:22:59.8946430Z 	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:538)
2020-08-27T09:22:59.8946585Z 	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:451)
2020-08-27T09:22:59.8946746Z 	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
2020-08-27T09:22:59.8946902Z 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
2020-08-27T09:22:59.8969292Z 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
2020-08-27T09:22:59.8970065Z 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
2020-08-27T09:22:59.8970249Z 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
2020-08-27T09:22:59.8970401Z 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
2020-08-27T09:22:59.8970802Z Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method
```

--- a/.github/workflows/ci-actions.yml
+++ b/.github/workflows/ci-actions.yml
@@ -103,7 +103,7 @@ jobs:
     needs: build-jdk11
     timeout-minutes: 240
     env:
-      MAVEN_OPTS: -Xmx2048m
+      MAVEN_OPTS: -Xmx2048m -XX:MaxMetaspaceSize=500m
     strategy:
       fail-fast: false
       matrix:
@@ -173,7 +173,7 @@ jobs:
     runs-on: windows-latest
     timeout-minutes: 180
     env:
-      MAVEN_OPTS: -Xmx1408m
+      MAVEN_OPTS: -Xmx1408m  -XX:MaxMetaspaceSize=500m
 
     steps:
       - uses: actions/checkout@v2

--- a/.github/workflows/ci-actions.yml
+++ b/.github/workflows/ci-actions.yml
@@ -103,7 +103,7 @@ jobs:
     needs: build-jdk11
     timeout-minutes: 240
     env:
-      MAVEN_OPTS: -Xmx2048m -XX:MaxMetaspaceSize=500m
+      MAVEN_OPTS: -Xmx2048m -XX:MaxMetaspaceSize=1000m
     strategy:
       fail-fast: false
       matrix:
@@ -173,7 +173,7 @@ jobs:
     runs-on: windows-latest
     timeout-minutes: 180
     env:
-      MAVEN_OPTS: -Xmx1408m  -XX:MaxMetaspaceSize=500m
+      MAVEN_OPTS: -Xmx1408m  -XX:MaxMetaspaceSize=1000m
 
     steps:
       - uses: actions/checkout@v2

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -33,7 +33,7 @@
         <microprofile-reactive-streams-operators.version>1.0.1</microprofile-reactive-streams-operators.version>
         <microprofile-rest-client.version>1.4.1</microprofile-rest-client.version>
         <smallrye-common.version>1.3.0</smallrye-common.version>
-        <smallrye-config.version>1.8.5</smallrye-config.version>
+        <smallrye-config.version>1.8.6</smallrye-config.version>
         <smallrye-health.version>2.2.3</smallrye-health.version>
         <smallrye-metrics.version>2.4.2</smallrye-metrics.version>
         <smallrye-open-api.version>2.0.7</smallrye-open-api.version>

--- a/core/builder/src/main/java/io/quarkus/builder/Execution.java
+++ b/core/builder/src/main/java/io/quarkus/builder/Execution.java
@@ -52,6 +52,7 @@ final class Execution {
         this.multis = new ConcurrentHashMap<>(builder.getInitialMulti());
         this.finalIds = finalIds;
         final EnhancedQueueExecutor.Builder executorBuilder = new EnhancedQueueExecutor.Builder();
+        executorBuilder.setRegisterMBean(false);
         executorBuilder.setCorePoolSize(8).setMaximumPoolSize(1024);
         executorBuilder.setExceptionHandler(JBossExecutors.loggingExceptionHandler());
         executorBuilder.setThreadFactory(new JBossThreadFactory(new ThreadGroup("build group"), Boolean.FALSE, null, "build-%t",

--- a/core/processor/src/main/java/io/quarkus/annotation/processor/ExtensionAnnotationProcessor.java
+++ b/core/processor/src/main/java/io/quarkus/annotation/processor/ExtensionAnnotationProcessor.java
@@ -100,11 +100,15 @@ public class ExtensionAnnotationProcessor extends AbstractProcessor {
 
     @Override
     public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
-        doProcess(annotations, roundEnv);
-        if (roundEnv.processingOver()) {
-            doFinish();
+        try {
+            doProcess(annotations, roundEnv);
+            if (roundEnv.processingOver()) {
+                doFinish();
+            }
+            return true;
+        } finally {
+            JDeparser.dropCaches();
         }
-        return true;
     }
 
     @Override

--- a/devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java
@@ -135,7 +135,6 @@ public class BuildMojo extends AbstractMojo {
 
     @Override
     public void execute() throws MojoExecutionException {
-
         if (skip) {
             getLog().info("Skipping Quarkus build");
             return;
@@ -201,6 +200,7 @@ public class BuildMojo extends AbstractMojo {
             QuarkusBootstrap.Builder builder = QuarkusBootstrap.builder()
                     .setAppArtifact(appArtifact)
                     .setMavenArtifactResolver(resolver)
+                    .setIsolateDeployment(true)
                     .setBaseClassLoader(BuildMojo.class.getClassLoader())
                     .setBuildSystemProperties(effectiveProperties)
                     .setLocalProjectDiscovery(false)

--- a/extensions/jdbc/jdbc-mysql/deployment/src/main/java/io/quarkus/jdbc/mysql/deployment/JDBCMySQLProcessor.java
+++ b/extensions/jdbc/jdbc-mysql/deployment/src/main/java/io/quarkus/jdbc/mysql/deployment/JDBCMySQLProcessor.java
@@ -33,6 +33,7 @@ import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
 import io.quarkus.deployment.builditem.NativeImageEnableAllCharsetsBuildItem;
 import io.quarkus.deployment.builditem.NativeImageEnableAllTimeZonesBuildItem;
+import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
 import io.quarkus.deployment.builditem.SslNativeConfigBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
@@ -66,8 +67,8 @@ public class JDBCMySQLProcessor {
 
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
-    void abandonedConnectionCleanUp(MySQLRecorder recorder) {
-        recorder.startAbandonedConnectionCleanup();
+    void abandonedConnectionCleanUp(MySQLRecorder recorder, ShutdownContextBuildItem shutdownContextBuildItem) {
+        recorder.startAbandonedConnectionCleanup(shutdownContextBuildItem);
     }
 
     @BuildStep

--- a/extensions/jdbc/jdbc-mysql/runtime/src/main/java/io/quarkus/jdbc/mysql/runtime/MySQLRecorder.java
+++ b/extensions/jdbc/jdbc-mysql/runtime/src/main/java/io/quarkus/jdbc/mysql/runtime/MySQLRecorder.java
@@ -2,15 +2,24 @@ package io.quarkus.jdbc.mysql.runtime;
 
 import org.graalvm.nativeimage.ImageInfo;
 
+import com.mysql.cj.jdbc.AbandonedConnectionCleanupThread;
+
 import io.quarkus.jdbc.mysql.runtime.graal.AbandonedConnectionCleanupThreadSubstitutions;
+import io.quarkus.runtime.ShutdownContext;
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class MySQLRecorder {
 
-    public void startAbandonedConnectionCleanup() {
+    public void startAbandonedConnectionCleanup(ShutdownContext shutdownContext) {
         if (ImageInfo.inImageRuntimeCode()) {
             AbandonedConnectionCleanupThreadSubstitutions.startCleanUp();
         }
+        shutdownContext.addShutdownTask(new Runnable() {
+            @Override
+            public void run() {
+                AbandonedConnectionCleanupThread.uncheckedShutdown();
+            }
+        });
     }
 }

--- a/extensions/jdbc/jdbc-mysql/runtime/src/main/java/io/quarkus/jdbc/mysql/runtime/graal/AbandonedConnectionCleanupThreadSubstitutions.java
+++ b/extensions/jdbc/jdbc-mysql/runtime/src/main/java/io/quarkus/jdbc/mysql/runtime/graal/AbandonedConnectionCleanupThreadSubstitutions.java
@@ -73,6 +73,11 @@ final public class AbandonedConnectionCleanupThreadSubstitutions implements Runn
         }
     }
 
+    @Substitute
+    public static void uncheckedShutdown() {
+        executorService.shutdownNow();
+    }
+
     private static void finalizeResourceAndRemoveReference(ConnectionFinalizerPhantomReference reference) {
         try {
             reference.finalizeResources();

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/DriverRemover.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/DriverRemover.java
@@ -1,0 +1,25 @@
+package io.quarkus.bootstrap.classloading;
+
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Enumeration;
+import org.jboss.logging.Logger;
+
+public class DriverRemover implements Runnable {
+
+    private static final Logger log = Logger.getLogger(DriverRemover.class);
+
+    @Override
+    public void run() {
+        Enumeration<Driver> drivers = DriverManager.getDrivers();
+        while (drivers.hasMoreElements()) {
+            Driver driver = drivers.nextElement();
+            try {
+                DriverManager.deregisterDriver(driver);
+            } catch (SQLException t) {
+                log.error("Failed to deregister driver", t);
+            }
+        }
+    }
+}

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java
@@ -17,6 +17,7 @@ import java.util.HashSet;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.ResourceBundle;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
@@ -81,6 +82,8 @@ public class QuarkusClassLoader extends ClassLoader implements Closeable {
         }
         PLATFORM_CLASS_LOADER = cl;
     }
+
+    private boolean closed;
 
     private QuarkusClassLoader(Builder builder) {
         //we need the parent to be null
@@ -461,6 +464,21 @@ public class QuarkusClassLoader extends ClassLoader implements Closeable {
 
     @Override
     public void close() {
+        synchronized (this) {
+            if (closed) {
+                return;
+            }
+            closed = true;
+        }
+        //DriverManager only lets you remove drivers with the same CL as the caller
+        //so we need do define the cleaner in this class loader
+        try (InputStream is = getClass().getResourceAsStream("DriverRemover.class")) {
+            byte[] data = JarClassPathElement.readStreamContents(is);
+            Runnable r = (Runnable) defineClass(DriverRemover.class.getName(), data, 0, data.length).newInstance();
+            r.run();
+        } catch (Exception e) {
+            log.debug("Failed to clean up DB drivers");
+        }
         for (ClassPathElement element : elements) {
             //note that this is a 'soft' close
             //all resources are closed, however the CL can still be used
@@ -471,6 +489,8 @@ public class QuarkusClassLoader extends ClassLoader implements Closeable {
                 log.error("Failed to close " + element, e);
             }
         }
+        ResourceBundle.clearCache(this);
+
     }
 
     @Override

--- a/integration-tests/resteasy-mutiny/src/test/java/io/quarkus/it/resteasy/mutiny/MutinyTest.java
+++ b/integration-tests/resteasy-mutiny/src/test/java/io/quarkus/it/resteasy/mutiny/MutinyTest.java
@@ -1,11 +1,12 @@
 package io.quarkus.it.resteasy.mutiny;
 
 import static io.restassured.RestAssured.get;
-import static org.awaitility.Awaitility.await;
 import static org.hamcrest.CoreMatchers.is;
 
+import java.time.Duration;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.function.Consumer;
 
 import javax.ws.rs.client.Client;
 import javax.ws.rs.client.ClientBuilder;
@@ -19,6 +20,8 @@ import org.junit.jupiter.api.Test;
 
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.subscription.UniEmitter;
 
 @QuarkusTest
 public class MutinyTest {
@@ -84,23 +87,32 @@ public class MutinyTest {
     public void testSSE() {
         Client client = ClientBuilder.newClient();
         WebTarget target = client.target("http://localhost:" + RestAssured.port + "/mutiny/pets");
-        SseEventSource source = SseEventSource.target(target).build();
-        List<Pet> pets = new CopyOnWriteArrayList<>();
-        try (SseEventSource eventSource = source) {
-            eventSource.register(event -> {
-                Pet pet = event.readData(Pet.class, MediaType.APPLICATION_JSON_TYPE);
-                pets.add(pet);
-            }, ex -> {
-                throw new IllegalStateException("SSE failure", ex);
+        try (SseEventSource eventSource = SseEventSource.target(target).build()) {
+            Uni<List<Pet>> petList = Uni.createFrom().emitter(new Consumer<UniEmitter<? super List<Pet>>>() {
+                @Override
+                public void accept(UniEmitter<? super List<Pet>> uniEmitter) {
+                    List<Pet> pets = new CopyOnWriteArrayList<>();
+                    eventSource.register(event -> {
+                        Pet pet = event.readData(Pet.class, MediaType.APPLICATION_JSON_TYPE);
+                        pets.add(pet);
+                        if (pets.size() == 5) {
+                            uniEmitter.complete(pets);
+                        }
+                    }, ex -> {
+                        uniEmitter.fail(new IllegalStateException("SSE failure", ex));
+                    });
+                    eventSource.open();
+
+                }
             });
-            eventSource.open();
-            await().until(() -> pets.size() == 5);
+            List<Pet> pets = petList.await().atMost(Duration.ofMinutes(1));
+            Assertions.assertEquals(5, pets.size());
+            Assertions.assertEquals("neo", pets.get(0).getName());
+            Assertions.assertEquals("indy", pets.get(1).getName());
+            Assertions.assertEquals("plume", pets.get(2).getName());
+            Assertions.assertEquals("titi", pets.get(3).getName());
+            Assertions.assertEquals("rex", pets.get(4).getName());
         }
-        Assertions.assertEquals("neo", pets.get(0).getName());
-        Assertions.assertEquals("indy", pets.get(1).getName());
-        Assertions.assertEquals("plume", pets.get(2).getName());
-        Assertions.assertEquals("titi", pets.get(3).getName());
-        Assertions.assertEquals("rex", pets.get(4).getName());
     }
 
     @Test

